### PR TITLE
HBASE-25622 Result#compareResults should compare tags.

### PR DESCRIPTION
**Pull request by shahrs87**
https://issues.apache.org/jira/browse/HBASE-25622

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   6m 41s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 26s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   7m 56s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   1m 19s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   1m 31s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   2m 46s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  1s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 21s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   1m 33s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   2m 45s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m 22s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 52s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 20s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   1m 20s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 33s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   1m 33s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 31s |  hbase-server: The patch generated 1 new + 3 unchanged - 0 fixed = 4 total (was 3)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 53s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 39s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   1m 10s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   1m 30s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   5m 40s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 56s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 43s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 103m 57s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 12s |  The patch does not generate ASF License warnings.  |
|  |   | 172m 38s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3004/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3004 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 1010c9f5e231 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-agent/workspace/Base-PreCommit-GitHub-PR_PR-3004/out/precommit/personality/provided.sh |
| git revision | branch-1 / 4cfbf19 |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3004/1/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3004/1/testReport/ |
| Max. process+thread count | 3623 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3004/1/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by shahrs87** [comment]
@virajjasani  could you please help review this small change ?

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 54s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 29s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   8m 58s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   1m 22s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   1m 36s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   2m 49s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  5s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 25s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   1m 32s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   2m 58s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m 45s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  7s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 22s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   1m 22s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 36s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   1m 36s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m 45s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m  3s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 53s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   1m 14s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   1m 34s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   6m  5s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 58s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 43s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 105m 15s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 16s |  The patch does not generate ASF License warnings.  |
|  |   | 171m 25s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3004/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3004 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 6d78354c7a6a 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-agent/workspace/Base-PreCommit-GitHub-PR_PR-3004/out/precommit/personality/provided.sh |
| git revision | branch-1 / 4cfbf19 |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3004/2/testReport/ |
| Max. process+thread count | 3214 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3004/2/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-common/src/main/java/org/apache/hadoop/hbase/CellUtil.java [hunk 1]
```diff
@@ -470,6 +470,14 @@ public static boolean matchingValue(final Cell left, final byte[] buf) {
         buf.length);
   }
 
+  /*
+    Compares tags in both cells.
+   */
+  public static boolean matchingTags(final Cell left, final Cell right) {
```
Comment (anoopsjohn) [inline hbase-common/src/main/java/org/apache/hadoop/hbase/CellUtil.java:476]: Seems this is used by verifyReplication only.  Still better follow what other matchingXXX do

Comment (shahrs87) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/CellUtil.java:476]: @anoopsjohn  Thanks for the review.
I did follow the same convention that [CellUtil#matchingValue](https://github.com/apache/hbase/blob/branch-1/hbase-common/src/main/java/org/apache/hadoop/hbase/CellUtil.java#L463-L466) does.
```
  public static boolean matchingValue(final Cell left, final Cell right) {
    return Bytes.equals(left.getValueArray(), left.getValueOffset(), left.getValueLength(),
        right.getValueArray(), right.getValueOffset(), right.getValueLength());
  }
```

> Still better follow what other matchingXXX do

Anoop,
could you please elaborate what you mean ?

Comment (anoopsjohn) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/CellUtil.java:476]: Ah.. This PR is against branch-1..  I thought its for master.. My comment is applicable for master.  Better u can create PR for master first and then other branches.  Can u open PR for master also pls?  Same approach will be applicable for branch-2 also.
So as such for this PR +1

Comment (shahrs87) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/CellUtil.java:476]: Tags are little bit convoluted in master branch. There are 2 implementation for KeyValue there..one backed by ByteBuffer and other backed by Array. We wanted this in our internal branch which is based of branch-1.6 so I created first again branch-1. Wanted to get a quick feedback from community first.

@anoopsjohn  Will create a PR against master soon.

**Pull request merged**